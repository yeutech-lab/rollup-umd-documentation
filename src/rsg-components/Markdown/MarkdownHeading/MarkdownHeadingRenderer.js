import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Heading from 'rsg-components/Heading';

const styles = ({ space }) => ({
	spacing: {
		marginBottom: space[2],
	},
});

function MarkdownHeadingRenderer({ classes, level, children }) {
	return (
		<div className={classes.spacing}>
			<Heading level={level}>{children}</Heading>
		</div>
	);
}

MarkdownHeadingRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
	children: PropTypes.node,
};

export default Styled(styles)(MarkdownHeadingRenderer);
