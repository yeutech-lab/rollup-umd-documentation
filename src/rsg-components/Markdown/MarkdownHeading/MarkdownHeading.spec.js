import React from 'react';
import { html } from 'cheerio';
import MarkdownHeading from './index';

describe('Markdown Heading', () => {
	it('should render a heading with a wrapper that provides margin', () => {
		const actual = render(<MarkdownHeading level={2}>The markdown heading</MarkdownHeading>);

		expect(html(actual)).toMatchSnapshot();
	});
});
