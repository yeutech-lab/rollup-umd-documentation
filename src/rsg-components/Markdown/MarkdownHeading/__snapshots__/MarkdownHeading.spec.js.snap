// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Heading should render a heading with a wrapper that provides margin 1`] = `

<div class="rsg--spacing-0">
  <h2 class="rsg--heading-2 rsg--heading2-4">
    The markdown heading
  </h2>
</div>

`;
